docs/newrates.ts

```diff
@@ -0,0 +1,200 @@
+ *  ],
+ *  "fiat": [{"fiatCode": "GBP"}, {"fiatCode": "EUR"}]
+ * }
+ *
+ * Response would be
+ * {
+ *   "targetFiat": "USD",
+ *   "crypto": [
+ *      {
+ *        "isoDate": "2025-06-23T15:52:30.000Z",
+ *        "pluginId": "bitcoin",
+ *        "rate": 103102.23
+ *      },
+ *      {
+ *        "isoDate": "2025-06-23T15:52:30.000Z",
+ *        "pluginId": "ethereum"
+ *        "rate": 2304.23
+ *      },
+ *      {
+ *        "isoDate": "2025-06-23T15:52:30.000Z",
+ *        "pluginId": "ethereum",
+ *        "tokenId": "dac17f958d2ee523a2206206994597c13d831ec7",
+ *        "rate": 1.002
+ *      }
+ *   ],
+ *   "fiat": [
+ *      {
+ *       "fiatCode": "GBP",
+ *       "rate": 1.32
+ *      }, {
+ *       "fiatCode": "EUR"
+ *       "rate": 1.08
+ *      }
+ *   ]
+ * }
+ */
+
+/**
+ * If no date is provided, rate queries should round down to the nearest minute.
+ * If a date is provided, rate queries should round down to the nearest 5 min interval
+ * since most historical rates are only accurate to 5 minutes
+ */
+
+/**
+ * Crypto rates are stored in the `rates_data`
+ * database.
+ *
+ * Doc id uses isoDate as the docId
+ */
+
+const ratesDocExample = {
+  _id: '2025-06-23T15:55:00.000Z',
+  chains: {
+    bitcoin: {
+      currencyCode: 'BTC',
+      USD: 101234.09
+    },
+    ethereum: {
+      currencyCode: 'ETH',
+      USD: 2310.09
+    }
+  },
+  tokens: {
+    dac17f958d2ee523a2206206994597c13d831ec7: {
+      currencyCode: 'USDT',
+      USD: 1.0001
+    },
+    a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48: {
+      currencyCode: 'USDC',
+      USD: 1.0002
+    }
+  },
+  fiat: {
+    CAD: { USD: 0.73 },
+    EUR: { USD: 1.15 },
+    GBP: { USD: 1.35 }
+  }
+}
+
+/**
+ * To prevent double querying assets that exist on
+ * multiple chains, a document will be used
+ * to map tokenId assets to another asset on a main chain
+ * like Ethereum.
+ *
+ * The examples below maps Arbitrum and Zksync to Ethereum and
+ * USDT/USDC on Solana to the same assets on Ethereum.
+ *
+ * These docs will be in the `rates_mappings` database
+ */
+
+// PluginID mappings should be manually edited, not autogenerated since
+// these ONLY refer to Edge pluginIds and not partner slugs
+const pluginIdMappings = {
+  _id: 'pluginIdMappings',
+  data: {
+    // Map arbitrum and zksync to ethereum since they are the same asset
+    arbitrum: 'ethereum',
+    zksync: 'ethereum'
+  }
+}
+
+/**
+ * Token mappings should be auto generated from partner APIs. ie. the
+ * coingecko /api/v3/coins/list?include_platform=true endpoint will return all the
+ * contract addresses on all chains for each asset supported.
+ *
+ * The mappings below should be refreshed once per day
+ */
+const tokenMappings = {
+  _id: 'tokenMappings',
+  data: {
+    // USDT from solana to ethereum
+    Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB: {
+      sourceChain: 'solana', // Only used for easy visual reference
+      destChain: 'ethereum', // Only used for easy visual reference
```

Comment:
should we determine a pluginId ranking to determine which chain is considered the source or trust the coingecko api to remain constant and use the first platform in the array?

Reply:
What Paul, William, Matt said. mapping keep growing but always look up main db first, mapping 2nd.